src/figuresClasses.ts

```diff
@@ -1,11 +1,88 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  sides: { a: number; b: number; c: number };
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        'The longest side of a triangle must be less than a sum of two others',
+      );
+    }
+    this.color = color;
+    this.sides = { a, b, c };
+  }
+
+  getArea(): number {
+    const { a, b, c } = this.sides;
+    const p = (a + b + c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
+
+  color: 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+  radius: number;
 
-export class Circle implements Figure {}
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    if (radius <= 0) {
+      throw new Error('All sides must be greater than 0');
```

Comment:
The error message 'All sides must be greater than 0' is misleading for a circle. Consider changing it to 'Radius must be greater than 0'.